File: src/semana-2/Search-Insert-Position.cpp
#include <bits/stdc++.h>
using namespace std;
// https://leetcode.com/problems/search-insert-position/
// 35. Search Insert Position
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        vector<int>::iterator it;
        it = lower_bound(nums.begin(), nums.end(), target);
        return it - nums.begin();
    }
};
int main(){
    vector<int> nums;
    nums.push_back(1);
    nums.push_back(3);
    nums.push_back(5);
    nums.push_back(6);
    int target = 7;
    Solution s;
    cout << s.searchInsert(nums, target) << endl;

    return 0;
}

File: src/semana-2/JavaScript/Search-Insert-Position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 * https://leetcode.com/problems/search-insert-position/
 */

const nums = [1,3,5,6]; 
const target = 5;

var searchInsert = (nums, target) => {
  var init = 0;
  var end = nums.length - 1;
  var mid = Math.floor((end + init) / 2);  

  while (init <= end) {
    if (target > nums[mid]) {
      init = mid + 1;
    } else if(target < nums[mid]) {
      end = mid - 1;
    } else {
      return mid;
    }
    
    mid = Math.floor((end + init) / 2);
  }
  return init;
};

console.log(searchInsert(nums, target));